### ISSUE TITLE
Checkpoint loading doesn't work

### ISSUE THREAD

**Issue by wistuba** [ContinualAI/avalanche#1270]
After fixing the issue as described in #1269, I get `AttributeError: Can't get attribute 'NoneType' on <module 'builtins' (built-in)>` when trying to load the checkpoint. Can be reproduced with the same example.

This is happening when `torch.load(..., pickle_module=dill)` is called.

**Comment by AntonioCarta** [comment]
Maybe you forgot to import something? pickle (and dill) need all the imports of the objects you are deserializing.
You should also use `if __name__ == "__main__":`.

This looks like a serialization problem unrelated to avalanche.

**Comment by wistuba** [comment]
Sorry, I do not understand. I am running following example script: https://github.com/ContinualAI/avalanche/blob/master/examples/checkpointing.py
The only modification is done in L155-156 where the worker related stuff is removed, see #1269.

This script uses `if __name__ == "__main__":` and since the very same script created the checkpoint, all relevant imports happened before calling `load_checkpoint_if_exists`.

**Comment by AntonioCarta** [comment]
Are you on PyTorch 1.13.0? It has a known serialization bug. If you use 1.12.0 it should work.

**Comment by wistuba** [comment]
Downgrading solved the issue, would be great if the dependencies could be updated accordingly.

One minor comment regarding the example: I was assuming `args.checkpoint_at` allows to set the checkpoint to load and where to continue the experiment. Seems not to be the case. The last checkpoint is loaded and the experiment is continued. What's the purpose of that argument in the example?

**Comment by AntonioCarta** [comment]
> Downgrading solved the issue, would be great if the dependencies could be updated accordingly.

I think a warning is better here. Some users may still want to use 1.13 without checkpointing.

I'm not sure. @lrzpellegrini can you explain what `checkpoint_at` is doing?

> The last checkpoint is loaded and the experiment is continued

Keep in mind that the other checkpoints are still saved so you can recover them if needed.

**Comment by wistuba** [comment]
> > The last checkpoint is loaded and the experiment is continued
> 
> Keep in mind that the other checkpoints are still saved so you can recover them if needed.

Agreed, this is not a problem with checkpointing, it's more of a problem of the example itself.


> > Downgrading solved the issue, would be great if the dependencies could be updated accordingly.
> 
> I think a warning is better here. Some users may still want to use 1.13 without checkpointing.

An exception with a meaningful message could be raised as soon as 1.13 is used in combination with checkpointing.

**Comment by lrzpellegrini** [comment]
Hi, the `checkpoint_at` argument is only used to stop the experiment at a certain experience. It doesn't change the behavior of the checkpointing plugin. We can consider adding an easy way to load a specific checkpoint instead of the last one (should be easy to implement).

We can also add a runtime check to warn the users about the issues with 1.13.0. However, the issue *should* be fixed starting from PyTorch version 1.13.1 (I did not check, but the fix was already on the main branch).

**Comment by Melika-Ayoughi** [comment]
I'm on version 1.13.1+cu116, and the error still exists.

**Comment by AntonioCarta** [comment]
Fixed by #1338
